dask/bytes/local.py

```diff
@@ -0,0 +1,120 @@
+    if '*' in fn:
+        filenames = list(map(os.path.abspath, sorted(glob(fn))))
+        sample, first = read_bytes(filenames[0], delimiter, not_zero,
+                                blocksize, sample=True, compression=compression)
+        rest = [read_bytes(f, delimiter, not_zero, blocksize, sample=False,
+                           compression=compression)[1] for f in filenames[1:]]
+        return sample, [first] + rest
+    else:
+        if not os.path.exists(fn):
+            raise FileNotFoundError(fn)
+
+        if blocksize is None:
+            offsets = [0]
+        else:
+            size = getsize(fn, compression)
+
+            offsets = list(range(0, size, blocksize))
+            if not_zero:
+                offsets[0] = 1
+
+        token = tokenize(fn, delimiter, blocksize, not_zero, compression,
+                         os.path.getmtime(fn))
+
+        logger.debug("Read %d blocks of binary bytes from %s", len(offsets), fn)
+        f = delayed(read_block_from_file)
+
+        values = [f(fn, offset, blocksize, delimiter, compression,
+                    dask_key_name='read-file-block-%s-%d' % (token, offset))
+                  for offset in offsets]
+
+        if sample:
+            if isinstance(sample, int) and not isinstance(sample, bool):
+                nbytes = sample
+            else:
+                nbytes = 10000
+            sample = read_block_from_file(fn, 0, nbytes, None, compression)
+
+        return sample, values
+
+
+def read_block_from_file(fn, offset, length, delimiter, compression):
+    with open(fn, 'rb') as f:
+        if compression:
+            f = SeekableFile(f)
+            f = compress_files[compression](f)
+        result = read_block(f, offset, length, delimiter)
+        if compression:
+            f.close()
+    return result
+
+
+def open_files(path):
+    """ Open many files.  Return delayed objects.
+
+    See Also
+    --------
+    dask.bytes.core.open_files: User function
+    """
+    myopen = delayed(open)
+    filenames = sorted(glob(path))
+    return [myopen(fn, mode='rb', dask_key_name='open-%s' %
+                   tokenize(fn, os.path.getmtime(fn))) for fn in filenames]
+
+
+from . import core
+core._read_bytes['local'] = read_bytes
+core._open_files['local'] = open_files
+
+
+if sys.version_info[0] >= 3:
+    def open_text_files(path, encoding=system_encoding, errors='strict'):
+        """ Open many files in text mode.  Return delayed objects.
+
+        See Also
+        --------
+        dask.bytes.core.open_text_files: User function
+        """
+        myopen = delayed(open)
+        filenames = sorted(glob(path))
+        return [myopen(fn, encoding=encoding, errors=errors,
+                       dask_key_name='open-%s' %
+                       tokenize(fn, encoding, errors, os.path.getmtime(fn)))
+                for fn in filenames]
+
+    core._open_text_files['local'] = open_text_files
```

Comment:
In Python 2 `open` doesn't support encoding/errors so we just fall back on the logic in `core/open_text_files`, which is pretty good at taking a binary file and treating it like a text file

```python
In [2]: open?
Docstring:
open(name[, mode[, buffering]]) -> file object

Open a file using the file() type, returns a file object.  This is the
preferred way to open a file.  See file.__doc__ for further information.
Type:      builtin_function_or_method
```